pkg/user/apiserver/registry/user/etcd/etcd_test.go

```diff
@@ -0,0 +1,30 @@
+package etcd
+
+import (
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestKubeAdminUserValidation(t *testing.T) {
+	userToBeTested := "kube:admin"
+
+	err := validateUsername(userToBeTested)
+
+	require.Nil(t, err)
+}
+
+func TestInvalidUserUserValidation(t *testing.T) {
+	userToBeTested := "system:admin"
+
+	err := validateUsername(userToBeTested)
+
+	require.EqualValues(t, 422, err.Status().Code)
+}
+
+func TestValidUserValidation(t *testing.T) {
+	userToBeTested := "john"
+
+	err := validateUsername(userToBeTested)
+
+	require.Nil(t, err)
+}
```

Comment:
this is basically just testing the `ValidateUserNameField` function behavior, we don't need to do that here. Feel free to move this to the same module, use golang generated tests to do that (this should be only one `Test*` function)

Reply:
consider using table driven tests, i.e. dave cheney once write a nice blog post: https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

Reply:
Updated, thanks!